python/basix/finite_element.py

```diff
@@ -602,19 +607,24 @@ def create_element(
     Returns:
         A finite element.
     """
-    return FiniteElement(
-        _create_element(
-            family.value,
-            celltype.value,
-            degree,
-            lagrange_variant.value,
-            dpc_variant.value,
-            discontinuous,
-            dof_ordering if dof_ordering is not None else [],
-            np.dtype(dtype).char,
-        )
+    if dtype is None:
```

Comment:
Remove?

Reply:
@mscroggs

Reply:
This is needed to allow DOLFINx to change the default scalar type as in the comment above

Reply:
This all seems too implicit to me. I think DOLFINx should explicitly pass the type to Basix functions. Global defaults are not nice.

Reply:
ok, default_scalar_type is gone